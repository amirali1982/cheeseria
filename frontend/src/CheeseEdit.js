import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import {Button, Container, Form, FormGroup, Input, Label} from 'reactstrap';
import AppNavbar from './AppNavbar';

class CheeseEdit extends Component {

    emptyItem = {
        name: '',
        color: '',
        price: ''
    };

    constructor(props) {
        super(props);
        this.state = {
            item: this.emptyItem,
            selectedFiles: ''
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    async componentDidMount() {
        if (this.props.match.params.id !== 'new') {
            const cheese = await (await fetch(`/cheeseria/${this.props.match.params.id}`)).json();
            this.setState({item: cheese});
        }
    }

    handleChange(event) {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        let item = {...this.state.item};
        item[name] = value;
        this.setState({item, selectedFiles: event.target.files});
    }

    async handleSubmit(event) {
        event.preventDefault();
        let formData = new FormData();
        const {item} = this.state;
        const {selectedFiles} = this.state;
        formData.append('cheese', JSON.stringify(item));
        if (selectedFiles && selectedFiles.length > 0) {
            formData.append('file', selectedFiles[0]);
        }

        await fetch('/cheeseria' + (item.id ? '/' + item.id : ''), {
            method: (item.id) ? 'PUT' : 'POST',
            body: formData,
        });
        this.props.history.push('/cheeseria');
    }

    render() {
        const {item} = this.state;
        const title = <h2>{item.id ? 'Edit Cheese' : 'Add Cheese'}</h2>;

        return <div>
            <AppNavbar/>
            <Container>
                {title}
                <Form onSubmit={this.handleSubmit}>
                    <FormGroup>
                        <Label for="name">Name</Label>
                        <Input type="text" name="name" id="name" value={item.name || ''}
                               onChange={this.handleChange} autoComplete="name"/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="color">Color</Label>
                        <Input type="text" name="color" id="color" value={item.color || ''}
                               onChange={this.handleChange} autoComplete="color"/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="price">Price</Label>
                        <Input type="text" name="price" id="price" value={item.price || ''}
                               onChange={this.handleChange} autoComplete="price"/>
                    </FormGroup>
                    <br/>
                    <Label for="image">Image</Label>&nbsp;
                    <input type="file" name="file" id="file" onChange={this.handleChange}/>
                    <br/>
                    <br/>
                    <FormGroup>
                        <Button color="primary" type="submit">Save</Button>{' '}
                        <Button color="secondary" tag={Link} to="/cheeseria">Cancel</Button>
                    </FormGroup>
                </Form>
            </Container>
        </div>
    }
}

export default CheeseEdit;