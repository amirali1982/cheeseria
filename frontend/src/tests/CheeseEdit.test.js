import renderer from 'react-test-renderer';
import CheeseEdit from "../CheeseEdit";
import {MemoryRouter} from "react-router-dom";

it('Test Cheese Edit', () => {
    const component = renderer.create(
        <MemoryRouter>
            <CheeseEdit match={{params: {id: 1}, isExact: true}}/>
        </MemoryRouter>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

})