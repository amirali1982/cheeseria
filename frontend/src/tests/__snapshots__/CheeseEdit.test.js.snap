// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cheese Edit 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/cheeseria-ui"
      onClick={[Function]}
    >
      Home
    </a>
    <div
      className="float-right"
    >
      <a
        aria-label={null}
        className="btn btn-success"
        href="/cheeseria-ui/new"
        onClick={[Function]}
      >
        Add Cheese
      </a>
    </div>
  </nav>
  <div
    className="container"
  >
    <h2>
      Add Cheese
    </h2>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="name"
        >
          Name
        </label>
        <input
          autoComplete="name"
          className="form-control"
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="color"
        >
          Color
        </label>
        <input
          autoComplete="color"
          className="form-control"
          id="color"
          name="color"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className=""
          htmlFor="price"
        >
          Price
        </label>
        <input
          autoComplete="price"
          className="form-control"
          id="price"
          name="price"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <br />
      <label
        className=""
        htmlFor="image"
      >
        Image
      </label>
      Â 
      <input
        id="file"
        name="file"
        onChange={[Function]}
        type="file"
      />
      <br />
      <br />
      <div
        className="form-group"
      >
        <button
          aria-label={null}
          className="btn btn-primary"
          onClick={[Function]}
          type="submit"
        >
          Save
        </button>
         
        <a
          aria-label={null}
          className="btn btn-secondary"
          href="/cheeseria-ui"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
    </form>
  </div>
</div>
`;
